### PR TITLE
Restructured code and other small changes

### PR DESCRIPTION
**Pull request by ChiCheng45**
**Description of work**
Restructured the code so that it follows the src-layout as shown in https://setuptools.pypa.io/en/latest/userguide/package_discovery.html which is more standard than the previous layout. Previously I was having some issues getting the editable build to work, the restructuring fixes this.  Also included some changes which improve things a bit which includes:

- updated gitignore to ignore: distribution/packaging files and directories,  generated pyd extensions, pycharm related stuff
- moved package data to the manifest.in
- changed script to entry-points so that they can be run as if they were executables
- updated the github wheels.yml to use python -m build since python setup.py bdist_wheel is deprecated https://packaging.python.org/en/latest/discussions/setup-py-deprecated/

**Fixes**
Fixes MDANSE and MDANSE_GUI installation using editable mode

**To test**
Run pip install -e . in the MDANSE and MDANSE_GUI and checked mdanse.exe and mdanse_gui.exe run without crashing. Built wheels with python -m build and installed them in a different environment to check building is done correctly and checked  mdanse.exe and mdanse_gui.exe again.

### CODE REVIEW

File: MDANSE_GUI/pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools", "numpy", "pip", "wheel"]
+requires = ["setuptools", "numpy", "pip", "wheel", "netCDF4"]
```
Comment (MBartkowiakSTFC) [inline MDANSE_GUI/pyproject.toml:2]: Was it necessary to bring back netCDF4? It has been one of our objectives to drop it altogether and use only HDF5.

File: MDANSE_GUI/requirements.txt [hunk 2]
```diff
@@ -2,4 +2,5 @@ MDANSE
 vtk
 qtpy
 PyYAML
-PyQt6
\ No newline at end of file
+PyQt6
+netCDF4
```
Comment (MBartkowiakSTFC) [inline MDANSE_GUI/requirements.txt:6]: Same as above, I was hoping that we could have MDANSE_GUI without netCDF4.

File: MDANSE_GUI/pyproject.toml (hunk 1 above)
Comment (ChiCheng45) [reply MDANSE_GUI/pyproject.toml:2]: I added it back in as there were some import errors. Its still being used in MDANSE_GUI\PyQtGUI\pygenplot\models\data_tree_model.py.

```
(test) PS C:\Users\xcb63893\PycharmProjects\MDANSE\MDANSE_GUI\dist> mdanse_gui.exe                                 
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\xcb63893\AppData\Local\anaconda3\envs\test\Lib\site-packages\MDANSE_GUI\Scripts\mdanse_gui.py", line 20, in main
    from MDANSE_GUI.PyQtGUI.main import startGUI
  File "C:\Users\xcb63893\AppData\Local\anaconda3\envs\test\Lib\site-packages\MDANSE_GUI\PyQtGUI\main.py", line 21, in <module>
    from MDANSE_GUI.PyQtGUI.MainWindow import Main
  File "C:\Users\xcb63893\AppData\Local\anaconda3\envs\test\Lib\site-packages\MDANSE_GUI\PyQtGUI\MainWindow.py", line 69, in <module>
    from MDANSE_GUI.PyQtGUI.pygenplot.widgets.main_window import MainWindow
  File "C:\Users\xcb63893\AppData\Local\anaconda3\envs\test\Lib\site-packages\MDANSE_GUI\PyQtGUI\pygenplot\widgets\main_window.py", line 20, in <module>
    from MDANSE_GUI.PyQtGUI.pygenplot.models.data_tree_model import (
  File "C:\Users\xcb63893\AppData\Local\anaconda3\envs\test\Lib\site-packages\MDANSE_GUI\PyQtGUI\pygenplot\models\data_tree_model.py", line 20, in <module>
    import netCDF4
ModuleNotFoundError: No module named 'netCDF4'
```

**Review by MBartkowiakSTFC — APPROVED** [review]
Excellent work! Works on my computer, and should be a big improvement on Windows machines.

**Pull request merged**